kong/observability/tracing/propagation/injectors/_base.lua

```diff
@@ -148,8 +152,9 @@ function _INJECTOR:inject(inj_tracing_ctx)
   -- Extractors automatically set `trace_id_original_size` during extraction.
   local orig_size = inj_tracing_ctx.trace_id_original_size
   local allowed = self.trace_id_allowed_sizes
-  local new_trace_id_size = contains(allowed, orig_size) and orig_size
-      or allowed[1]
+  local lookup = self.trace_id_allowed_sizes_lookup
+
+  local new_trace_id_size = (orig_size ~= nil and lookup[orig_size]) and orig_size or allowed[1]
```

Comment:
Because table index can't be assigned if it's `nil`. So this can be simplified to `lookup[orig_size] and orig_size or allowed[1]`, which is more clean.

Reply:
update